### ISSUE TITLE
Reproduced results differ from paper and resume training from ckpt

### ISSUE THREAD

**Issue by TiminHu** [uzh-rpg/ssms_event_cameras#12]
Hello, I recently read your paper and congratulations on its selection for the CVPR spotlight. The methodology in your paper struck me as very innovative. Therefore, after reading your paper, I downloaded your code and tried to reproduce it myself. However, I encountered some issues during the reproduction process. I was wondering if you would be available to help when you have some free time.
1. I used the same experimental configuration on the Gen1 dataset as in the paper.
MDL_CFG=small
GPU_IDS=1
BATCH_SIZE_PER_GPU=8
TRAIN_WORKERS_PER_GPU=6
EVAL_WORKERS_PER_GPU=2
And I conducted the test on an A800 with 80G of memory. The performance after 400K training is as follows:
'val/AP': 0.42775878876024126, 
'val/AP_50': 0.6627249514964132, 
'val/AP_75': 0.45529512680521994, 
'val/AP_S': 0.3671580891247672, 
'val/AP_M': 0.49355016226953197, 
'val/AP_L': 0.2880353366142091
It seems that there is a significant difference from the results of the small-sized model in the paper. I also noticed the author's response to another issue and would like to ask whether the difference in results is related to the type of graphics card or to the inability to fix the random seed due to the streaming processing used in the paper?

2. In a previous training session, the model was interrupted at 80,000 iterations due to some reasons. Later, I wanted to resume the training. I set the command-line arguments for wandb.artifact_name, wandb.artifact_local_file, wandb.wandb_runpath, but when I restarted, the model first conducted a test. After the test explanation, there was the following error, which seems to be about a mismatch in the log keys. I would like to ask why this is happening?
"lightning.fabric. utilities.exceptions.isconfipurationexception: "odelcheckpoint(momnitor= val/Ap?') could not find the monitored key in the returned metrics: ['epoch', 'step'”. hr: id youcall`log('val/AP', value) in the LightningModule ?"

3. When I trained on Gen1 with the configuration from question 1, it took about 77 hours. Is this speed normal?

Looking forward to your reply.

**Comment by NikolaZubic** [comment]
Hi Timin, thanks for your questions. Sorry for the later reply.

1. We observed that there are problems when using different GPUs than A100 and had users report us that on different GPU when training the model sometimes they trained few times until they got the result we have. On A100 they were able to reproduce every time. Unfortunately, S5 model changes we used were very GPU-dependent and also "hacked" when converting it from JAX to Torch code, so this definitely can cause some instabilities. Further solution would be to integrate S5 Jax version into this code where you would have two computational graphs, one for S5 instances and Torch for other models, definitely can be done, you just have to take into account when you pass inputs to S5 that you convert them to jax tensors and back to torch tensors. Also, back-propagation should be done explicitly such that you are sure S5 weights also get updated, definitely doable.
2. Indeed a very weird error that happened to me before and hence I solved it by first doing the test because when model does the test, then it is able to fetch the validation AP. Why this is not sufficient with you I am not sure but the reason could be a newer version of PyTorch or some update they did recently.
3. Since on A100 training took around 2 days and A100 can be up to 35% faster than A800, that is totally okay, but should maybe be a little bit faster.
